src/QsCompiler/CompilationManager/AssemblyLoader.cs

```diff
@@ -172,6 +172,53 @@ public static bool LoadReferencedAssembly(Uri asm, out References.Headers header
             return compilation != null && IsValidCompilation(compilation);
         }
 
+        /// <summary>
+        /// Loads any QIR byte code included as a resource from <paramref name="assemblyFile"/>.
+        /// </summary>
+        /// <param name="assemblyFile">The reader for the byte stream of a dotnet DLL from which to load the compilation.</param>
+        /// <param name="qirByteCode">The QIR byte code included as a resource of <paramref name="assemblyFile"/>.</param>
+        /// <returns>
+        /// True if <paramref name="assemblyFile"/> includes a suitable resource, false otherwise.
+        /// </returns>
+        public static bool LoadQirByteCode(
+            PEReader assemblyFile,
+            [NotNullWhen(true)] out byte[]? qirByteCode)
+        {
+            qirByteCode = null;
+            var metadataReader = assemblyFile.GetMetadataReader();
+            bool isResourcePresent = false;
+            ManifestResource resource;
+            if (metadataReader.Resources().TryGetValue(DotnetCoreDll.ResourceNameQsDataQirV1, out resource))
+            {
+                isResourcePresent = true;
+            }
+
+            // The offset of resources is relative to the resources directory.
+            // It is possible that there is no offset given because a valid dll allows for extenal resources.
+            // In all Q# dlls there will be a resource with the specific name chosen by the compiler.
+            var resourceDir = assemblyFile.PEHeaders.CorHeader.ResourcesDirectory;
+            if (!assemblyFile.PEHeaders.TryGetDirectoryOffset(resourceDir, out var directoryOffset) ||
+                !isResourcePresent ||
+                !resource.Implementation.IsNil)
+            {
+                return false;
+            }
+
+            // This is going to be very slow, as it loads the entire assembly into a managed array, byte by byte.
+            // Due to the finite size of the managed array, that imposes a memory limitation of around 4GB.
+            // The other alternative would be to have an unsafe block, or to contribute a fix to PEMemoryBlock to expose a ReadOnlySpan.
+            PerformanceTracking.TaskStart(PerformanceTracking.Task.LoadDataFromReferenceToStream);
```

Comment:
No need to include performance tracking in this method since it is not used during compilation.

Reply:
removed